sdk/src/tools/code-search.ts

```diff
@@ -0,0 +1,80 @@
+import { spawn } from 'child_process'
+import * as path from 'path'
+
+import { rgPath } from '@vscode/ripgrep'
+
+import type { CodebuffToolOutput } from '../../../common/src/tools/list'
+
+export function codeSearch({
+  pattern,
+  flags,
+  cwd,
+}: {
+  pattern: string
+  flags?: string
+  cwd: string
+}): Promise<CodebuffToolOutput<'code_search'>> {
+
+  return new Promise((resolve) => {
+    let stdout = ''
+    let stderr = ''
+
+    const flagsArray = (flags || '').split(' ').filter(Boolean)
+    let searchCwd = cwd
+    
+    // Note: In the SDK, we don't have access to a project root concept,
```

Comment:
The client object has a this.cwd, which is like the project root. The cwd from the tool call should be concatenated with that!